File: src/main/java/money/scheduler/model/PaymentYear.java
package money.scheduler.model;

import java.util.LinkedList;
import java.util.List;

public class PaymentYear {
    private int year;
    private List<Month> months;

    public PaymentYear(int year, List<Month> months) {
        this.year = year;
        this.months = months;
    }

    public static PaymentYear of(int year, List<Integer> months) {
        List<Month> rawMonths = new LinkedList<>();
        for (int i=1; i<=12; i++) {
            rawMonths.add(new Month(i, months.contains(i)));
        }
        return new PaymentYear(year, rawMonths);
    }

    public int getYear() {
        return year;
    }

    public List<Month> getMonths() {
        return months;
    }
}

File: src/main/java/money/scheduler/model/PaymentYear.kt
package money.scheduler.model

import java.util.LinkedList

class PaymentYear(val year: Int, val months: List<Month>) {

    companion object {
        fun of(year: Int, months: List<Int?>): PaymentYear {
            val rawMonths: MutableList<Month> = LinkedList()
            for (i in 1..12) {
                rawMonths.add(Month(i, months.contains(i)))
            }
            return PaymentYear(year, rawMonths)
        }
    }
}